### ISSUE TITLE
[a11y]: Unable to have an interactive element with role="button" inside DefinitionTooltip

### ISSUE THREAD

**Issue by shreevidya-ibm** [carbon-design-system/carbon#18266]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

v1.62.1

### React version

v18.2.0

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker - latest deployment

### Assistive technology

MacOS VoiceOver

### Description

Description
There is an accessibility issue when attempting to use an interactive element, such as a button with `role="button"`, inside the `DefinitionTooltip` component. The DefinitionTooltip implementation does not seem to support or accommodate interactive child elements. 

Recommended fix:
Interactive elements (e.g., < Button role="button" >) should be allowed and functional inside a DefinitionTooltip component.

### Reproduction/example

https://stackblitz.com/edit/github-qktfqi5o?file=src%2FApp.jsx

### Steps to reproduce

- Open the Target Web Page
- Open Developer Tools
- Navigate to the Accessibility Assessment Tab
- Run the IBM Equal Access Accessibility Checker
- Analyze the Violation stating 'The element with role "button" contains descendants with roles "button" which are ignored by browsers'

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by guidari** [comment]
Hey @shreevidya-ibm 
Tooltips do not support interactive elements. You can use Toggletip instead.
Docs: https://carbondesignsystem.com/components/tooltip/usage/#overview

**Comment by shreevidya-ibm** [comment]
Hi @guidari , I find the same issue with Toggletip as well as Tooltip. Please advise

**Comment by shreevidya-ibm** [comment]
Created a similar issue for Toggletip: https://github.com/carbon-design-system/carbon/issues/18499
